mcstatus/server.py

```diff
@@ -22,14 +23,28 @@
     from typing_extensions import Self
 
 
-__all__ = ["JavaServer", "BedrockServer", "MinecraftServer", "MinecraftBedrockServer"]
+__all__ = [
+    "NotBedrockAndNotJavaServer",
+    "MCServer",
+    "JavaServer",
+    "BedrockServer",
+    "MinecraftServer",
+    "MinecraftBedrockServer",
+]
+
+
+class NotBedrockAndNotJavaServer(TypeError):
+    """Exception for when server is not Bedrock or Java."""
 
 
 class MCServer(ABC):
-    """Base abstract class for a general minecraft server.
+    """Base class for a general minecraft server.
+
+    This class contains the basic logic shared across both java and bedrock versions,
+    also it to can be used as cross-platform ping.
 
-    This class only contains the basic logic shared across both java and bedrock versions,
-    it doesn't include any version specific settings and it can't be used to make any requests.
+    Firstly it will try to ping server as `JavaServer`, if it fails it will try to ping as `BedrockServer`.
+    You can reverse it with the `first_bedrock` argument.
```

Comment:
This should be reverted, as after the discussed changes, `MCServer` class will stay abstract and will not contain cross-platform behavior, also, even if it did, this is no longer the PR to do that in.

Reply:
Reverted in 5d2d6c4faba70fe7c726cc8ea756cef764a090cd.